lib/hotel.rb

```diff
@@ -0,0 +1,119 @@
+# hotel
+require_relative 'room.rb'
+require 'date'
+
+module HotelManagment
+  class Hotel
+    attr_reader :rooms, :reservations, :found_reservations, :unreserved_rooms, :blocks
+    def initialize
+      @rooms = []
+      @reservations = []
+      @found_reservations = []
+      @unreserved_rooms = []
+      @blocks = []
+    end
+
+    def add_20_rooms
+      room_number = 1
+      20.times do
+        room = HotelManagment::Room.new(room_number)
+        @rooms << room
+        room_number += 1
+      end
+    end
+
+
+    # creates a reservation
+    def create_reservation(first_name, last_name, check_in_date, check_out_date, room_number)
+
```

Comment:
The `Reservation` class should be checking for it being created with invalid dates.